File: Searching/Assignments/Assignment 1/Main.java
/**
 * @author Fredrik Lundström
 * 2021-09-29
 * Write a simple filter to clean a text, i.e. to 
 * remove all characters that are not alphabetic, blank 
 * or newline - replacing every such character by a 
 * blank to keep the number of characters constant to 
 * the original text.
 */
import java.util.*;

public class Main {
    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        String text = scanner.nextLine();
        scanner.close();

        char[] char_array = text.toCharArray();

        for(int i = 0; i < char_array.length; i++){
            if(isAlpha(char_array[i]) == false)
                char_array[i] = ' ';
        }
        
        text = String.valueOf(char_array);
        System.out.println("Text: " + text);
    }

    /**
     * This method returns the boolean value true if parameter c 
     * is alphabetical or false if not.
     * @param c is a character
     * @return true or false if c is alphabetical or non-alphabetical
     */
    private static Boolean isAlpha(char c){
        if((int)c >= 97 && (int)c <= 122 )
            return true;
        else
            return false;
    }
}

File: Searching/Assignments/Assignment 1/main.c
#include <stdio.h>
#include <stdlib.h>

/*
	@Author Fredrik Lundström
	2021-09-28
    Write a simple filter to clean a text, i.e. to remove all 
    characters that are not alphabetic, blank or newline - replacing 
    every such character by a blank to keep the number of characters 
    constant to the original text.
*/

int main(int argc, char *argv[]){
    int c;
    readfromfile(c);
}

/*
    Functino readfromfile reads the text from a file instead of input 
    input in the terminal and writes it in a new text file but with
    only alphabetical characters.
*/
void readfromfile(int c){
    FILE * file;
    file = fopen("thetext.txt", "r");

    FILE * newfile;
    newfile = fopen("onlyalpha.txt", "w");

    int i = 0;
    while((c = getc(file)) != EOF){
        if(isalpha(c) == 0)
            c = NULL;
        else
            fputc(c, newfile);
        i++;
    }
    fclose(file);
    fclose(newfile);
}

void readfromterminal(int c){
    while((c = getchar()) != '\n'){
        if(isalpha(c) == 0)
            c = NULL;
    }
    putchar(c);
}